src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1179,9 +1173,8 @@ void OrbitMainWindow::OnStadiaConnectionError(std::error_code error) {
                               .arg(target.GetConnection()->GetInstance().display_name)
                               .arg(QString::fromStdString(error.message()));
 
-  target_label_->setStyleSheet(
-      kTargetLabelDefaultStyleSheet.arg(kTargetLabelColorTargetDisconnected));
-  target_label_->setToolTip(error_message);
+  target_label_->SetConnectionDead(error_message);
+  ui->menuBar->adjustSize();
```

Comment:
Nit: This feels a bit like a dependency on the implemenation of TargetLabel, because you need to know which method of TargetLabel changes the widget's size. Ideally this is decoupled and TargetLabel tells you when a call of adjustSize is needed.

This could be done by a signal in TargetLabel. Maybe there is already one in QWidget which could be used. Haven't checked though.

Reply:
I like this idea a lot. QWidget did not have a signal that was useful, so I added a `SizeChanged` signal. done